lambda-layer/packages/layer/scripts/otel-instrument

```diff
@@ -1,5 +1,40 @@
 #!/bin/bash
-export NODE_OPTIONS="${NODE_OPTIONS} --require /opt/wrapper.js"
+isESMScript() {
+  # Lambda function root directory
+  TASK_DIR="/var/task"
+
+  # Flag variables to track conditions
+  local found_mjs=false
+  local is_module=false
+
+  # Check for any files ending with `.mjs`
+  if ls "$TASK_DIR"/*.mjs &>/dev/null; then
+      found_mjs=true
+    fi
+
+  # Check if `package.json` exists and if it contains `"type": "module"`
+  if [ -f "$TASK_DIR/package.json" ]; then
+    # Check for the `"type": "module"` attribute in `package.json`
+    if grep -q '"type": *"module"' "$TASK_DIR/package.json"; then
+      is_module=true
+    fi
+  fi
+
+  # Return true if both conditions are met
+  if $found_mjs || $is_module; then
+    return 0  # 0 in bash means true
+  else
+    return 1  # 1 in bash means false
+  fi
+}
+
+if isESMScript; then
```

Comment:
Do we still need this, given we have a new `otel-instrument-esm` script? Or vice versa?

Reply:
yes. `otel-instrument-esm` is the fallback solution if the esm auto-detection doesn't work. We want to provide out of box CX on ESM support with the best efforts.